controllers/sessionController.js

```diff
@@ -0,0 +1,49 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
```

Comment:
Nitpick:
Since you’re already using a try/catch, I’d recommend leveraging it for checking the passwords:
```js

const registerDo = async (req, res) => {
   try {
     if (req.body.password !== req.body.password1) {
       // throwing password error here gets caught below
       throw new Error("The passwords do not match.");
     }

     await User.create(req.body);
     res.redirect("/");
   } catch (e) {
     if (e.constructor.name === "ValidationError") {
       parseVErr(e, req);
     } else if (e.name === "MongoServerError" && e.code === 11000) {
       req.flash("error", "That email address is already registered.");
     } else {
       req.flash("error", e.message);
     }

     res.render("register", { errors: req.flash("error") });
   }
 };
```